### ISSUE TITLE
tproxy method causes lock up on fedora vanilla kernels 

### ISSUE THREAD

**Issue by colinmkeith** [sshuttle/sshuttle#278]
In trying to debug a problem with a blank screen on waking up my laptop I switched to the mainline kernels for Fedora 28. Since doing so running with tproxy method causes the entire machine to lock up.

Running the command

```./run -vvvvv --method=tproxy --dns -r work <network>```

I took a photo of the screen and it gets to the end of adding rules to the mangle table, sends a PONG cmd, receives a PING response, starts Waiting and then everything locks up:

```
>> iptables -t mangle -A sshuttle-t-12300 -j TPROXY --tproxy-mark 0x1/0x1 --dest <network> -m udp -p udp --on-port 12300
c : < channel=0 cmd=PONG len=7
c : receiving PING response
c : Waiting: 4 r[5, 8, 9, 10, 11, 12, 14] w=[] x=[] (fullness=0/0)
```

And then the entire computer has locked up. The HD activity light doesn't flash at all, only the RF indicator and the power light are on. No windows respond. I had journalctl running in another window and it displayed nothing before things locked up.

Fails like this with:

kernel-core-4.19.0-0.rc8.git0.1.vanilla.knurd.1.fc28.x86_64
kernel-core-4.19.0-0.rc5.git0.1.vanilla.knurd.1.fc28.x86_64

Has worked wonderfully well with standard fedora kernels, tested using latest copy of sshuttle from git (commit 97ed2030f340113908fa218f90964767415c55ab)

./run picks python3, which is version Python 3.6.6 (RPM: python3-3.6.6-1.fc28.x86_64)

Unfortunately I'm not sure how to debug this further.

Any helpful suggestions would be appreciated.

**Comment by colinmkeith** [comment]
Sorry, forgot to add that this works fine:
```./run -vvv --dns -r work <network>```

But it doesn't have UDP support unfortunately.

**Comment by brianmay** [comment]
Sounds nasty :-(. Suspect it may be a kernel issue, not sshuttle. Even if sshuttle installed iptables rules that broke all IP communications, I don't think it should be making your computer unusable quite like this.

All I can suggest is maybe a serial console running something like journalctl or "dmesg -w" - you might find the system crashes before it can update the X-Window with the logging information.

**Comment by colinmkeith** [comment]
Thanks for the suggestion. I booted into run level 3 but had difficulty with the network service failing to start because of the config for the wired ethernet port set to ONBOOT=yes which caused systemd to try and bring up networking on that interface and fail. I set it to ONBOOT=no and tried running sshuttle as above with the tproxy method and it worked. So I rebooted back into run level 5 and sshuttle set up the SSH tunnel as normal.

I am at a complete loss as to explain what is happenin here as the only change in the whole system was setting the ONBOOT flag to no on the wired ethernet interface.

Either way, this is clearly not related to sshuttle doing anything special once the iptables rules are added, so I'm closing out this issue. Thanks again.
